zebra-cli/src/commands/getnewaddress.rs

```diff
@@ -0,0 +1,114 @@
+//! `getnewaddress` subcommand
+
+use std::convert::TryInto;
+
+use abscissa_core::{config, Command, FrameworkError, Options, Runnable};
+use qrcode::{render::unicode, QrCode};
+use zcash_address::{self, unified, ToAddress};
+
+use zebra_chain::{orchard, sapling};
+
+use crate::config::ZebraCliConfig;
+
+/// `getnewaddress` subcommand
+///
+/// The `Options` proc macro generates an option parser based on the struct
+/// definition, and is defined in the `gumdrop` crate. See their documentation
+/// for a more comprehensive example:
+///
+/// <https://docs.rs/gumdrop/>
+#[derive(Command, Debug, Options)]
+pub struct GetNewAddressCmd {
+    // Example `--foobar` (with short `-f` argument)
+// #[options(short = "f", help = "foobar path"]
+// foobar: Option<PathBuf>
+
+// Example `--baz` argument with no short version
+// #[options(no_short, help = "baz path")]
+// baz: Options<PathBuf>
+
+// "free" arguments don't have an associated flag
+// #[options(free)]
+// free_args: Vec<String>,
+}
+
+impl config::Override<ZebraCliConfig> for GetNewAddressCmd {
+    // Process the given command line options, overriding settings from
+    // a configuration file using explicit flags taken from command-line
+    // arguments.
+    fn override_config(&self, config: ZebraCliConfig) -> Result<ZebraCliConfig, FrameworkError> {
+        Ok(config)
+    }
+}
+
+impl Runnable for GetNewAddressCmd {
+    /// Start the application.
+    fn run(&self) {
+        let network = zebra_chain::parameters::Network::Mainnet;
```

Comment:
Optional idea: not sure if this would actually help with readability, but I thought it could help to separate this into different methods:

```rust
fn run(&self) {
    let network = zebra_chain::parameters::Network::Mainnet;

    let sapling_address = self.new_sapling_address(network);
    let orchard_address = self.new_orchard_address(network);

    let zcash_address =
        self.new_unified_address(network, vec![sapling_address, orchard_address]);

    let qr_code = self.create_qr_code_image(&zcash_address);

    println!("\nNew Unified Zcash Address:");
    println!("\n{}\n", zcash_address);
    println!("\n{}\n", qr_code);
}
```

I think the `network` parameters could also be omitted if it's stored in `GetNewAddressCmd` as a configuration option, but maybe that's not necessary for now :thinking: